wifi_password/utils.py

```diff
@@ -0,0 +1,183 @@
+
+import constants
+
+def get_platform() -> str:
+    """
+    Returns the name of the platform where the application is currently running
+    """
+    platforms = {
+        'linux1': constants.LINUX,
+        'linux2': constants.LINUX,
+        'darwin': constants.MAC,
+        'win32': constants.WINDOWS
+    }
+
+    if not sys.platform in platforms:
+        return sys.platform
+
+    return platforms[sys.platform]
+
+def get_profiles() -> list:
+    """
+    Gets a list of names from saved wifi networks in the current platform
+    """
+    profiles = []
+
+    platform = get_platform()
+
+    try:
+        if platform == constants.MAC:
+            pass
+        elif platform == constants.LINUX:
+            pass
+        elif platform == constants.WINDOWS:
+            # Reference: https://www.geeksforgeeks.org/getting-saved-wifi-passwords-using-python/
+            # getting meta data
+            meta_data = run_command('netsh wlan show profiles | findstr ":"')
+
+            if meta_data != "":
+                # splitting data line by line
+                data = meta_data.split('\n')
+
+                if len(data) > 0:
+                    # skip the first element in data since it does not contain a valid profile returned by netsh command
+                    profiles = [d.split(':')[1].strip(' .\r') for d in data[1:]]
+    except Exception as ex:
+        print(f'Error: {ex}')
+
+    return profiles
+
+def generate_wifi_dict(profiles: list) -> dict:
+    """
+    Generates a dictionary with the wifi name as key and the password as it's value
+    """
+    wifi_dict = {}
+
+    if profiles is None:
+        print(f'List is not defined.')
+        return
+
+    if len(profiles) == 0:
+        print(f'List is empty.')
+        return
+
+    for ssid in profiles:
+        password = get_password(ssid)
+
+        wifi_dict[ssid] = password
+
+    return wifi_dict
+
+def get_password(ssid: str) -> str:
+    """
+    Gets the password for a given SSID
+    """
+    password = ""
+
+    if ssid == "" or ssid is None:
+        print("SSID is not defined")
+        return password
+
+    platform = get_platform()
+
+    try:
+        if platform == constants.MAC:
+            password = run_command(f"security find-generic-password -l \"{ssid}\" -D 'AirPort network password' -w")
+        elif platform == constants.LINUX:
+            # Check if the user is running with super user privilages
+            if os.getuid() != 0:
+                password = run_command(f"sudo nmcli -s -g 802-11-wireless-security.psk connection show '{ssid}'")
+            else:
+                password = run_command(f"nmcli -s -g 802-11-wireless-security.psk connection show '{ssid}'")
+        elif platform == constants.WINDOWS:
+            password = run_command(f"chcp 437 | netsh wlan show profile name=\"{ssid}\" key=clear | findstr Key")
+
+            if password != "":
+                password = re.findall(r"Key Content\s+:\s(.*)", password)[0]
```

Comment:
I think here you can return an empty string for a password-free profile instead of printing an error.

```suggestion
            password = re.findall(r"Key Content\s+:\s(.*)", password)

            if len(password) > 0:
                password = password[0]
            else:
                password = ""
```

Reply:
Hi,

I just pushed the changes as suggested.